src/app/InteractionModelEngine.cpp

```diff
@@ -370,8 +360,27 @@ bool InteractionModelEngine::SubjectHasActiveSubscription(FabricIndex aFabricInd
 
 bool InteractionModelEngine::SubjectHasPersistedSubscription(FabricIndex aFabricIndex, NodeId subjectID)
 {
-    // TODO(#30281) : Implement persisted sub check and verify how persistent subscriptions affects this at ICDManager::Init
-    return false;
+    bool persistedSubMatches = false;
+
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+    auto * iterator = mpSubscriptionResumptionStorage->IterateSubscriptions();
+    // Verify that we were able t0 allocate an iterator. If not, assume we have a persisted subscription.
```

Comment:
Why does this assume we have a persisted subscription on failure?  Won't this lead to failure to check in (because we expect the persisted subscription to do it), which could lead to a device that can't be communicated with?

Seems safer to assume that there is no match, and document why we are making that assumption...

Reply:
That is a valid approach. I had opted with sending the least of messages possible. If we don't know if there will be a subscription resumption attempt, assume one will be present which will at worst delay the recovery by one IdleModeDuration but save some battery life.

Reply:
OK.  If we are guaranteed that the delay is at most one IdleModeDuration (not "forever") that seems like a reasonable tradeoff, that should be documented in the code.